File: UI/src/main/java/ca/warp7/rt/feature/ast/ASTFeature.java
package ca.warp7.rt.feature.ast;

import ca.warp7.rt.core.feature.Feature;
import ca.warp7.rt.core.feature.FeatureItemTab;
import ca.warp7.rt.core.feature.FeatureUtils;
import javafx.scene.Parent;

import java.util.Collections;
import java.util.List;

public class ASTFeature implements Feature {

    @Override
    public List<FeatureItemTab> getLoadedTabs() {
        return Collections.singletonList(new FeatureItemTab(
                "Alliance Selection", "fas-list-alt:18:gray", getFeatureId(),
                FeatureItemTab.Group.SingleTab, ""
        ));
    }

    @Override
    public String getFeatureId() {
        return "ast";
    }

    @Override
    public Parent onOpenTab(FeatureItemTab tab) {
        return FeatureUtils.loadParent("/ca/warp7/rt/feature/ast/AST.fxml");
    }
}

File: app/src/main/kotlin/ca/warp7/rt/ext/ast/ASTFeature.kt
package ca.warp7.rt.ext.ast

import ca.warp7.rt.api.Feature
import ca.warp7.rt.api.FeatureLink
import ca.warp7.rt.api.loadParent
import javafx.scene.Parent

class ASTFeature : Feature {
    override val link = FeatureLink("Alliance Selection", "fas-tasks", 18)

    override fun onOpen(): Pair<Parent?, Parent?> {
        return null to loadParent("/ca/warp7/rt/ext/ast/AST.fxml")
    }
}
